### DISCUSSION TITLE
How to train a multigraph network

### DISCUSSION THREAD

**Discussion by john-como** [pyg-team/pytorch_geometric#7664] [Q&A]
Hello all! I am curious what it would look like to train a multigraph network where the graphs are time related. For instance, we may want to forecast demand on a product for the next _k_ intervals. I can create _N_ such graphs, each being a bipartite graph representing users and products representing interactions over a given time length, but am unsure of how to set up training.  Ideally as well, these learned network embeddings would serve as input in a TCN or other time related model (GRU, RNN, etc.). I understand conceptually how this works, as in training all the networks for a given task (link prediction is most suitable here) and the sequential models and backprop the loss all the way. I am having trouble formalizing this in code however. Any pointers would be great! Thanks! 

Inspiration for this question is stemmed in the research article below: 
https://assets.amazon.science/50/90/df9385f840c7b0363febf882a6ad/spatio-temporal-multi-graph-networks-fordemand-forecasting-in-online-marketplaces.pdf

**Comment by wsad1** [comment]
Here are some examples that might help you
1. https://github.com/pyg-team/pytorch_geometric/blob/master/examples/tgn.py. This uses a TGN to model a temporal graph problem. Here both nodes, edges and features can be added/removed/updated over time.
2. Pytorch Geometric Temporal, is a library built on top of PyG for modelling temporal graphs. Some [examples](https://pytorch-geometric-temporal.readthedocs.io/en/latest/notes/introduction.html#applications) from there might be useful too.

**Reply by john-como** [reply]
Thank you! I looked into Torch's Neuro Spatiotemporal but did not come across this, it looks much more applicable thanks again! Will leave thread not answered still for others!
